### DISCUSSION TITLE
Should ui.tab_panels be removed from its strong dependency on ui.tabs?

### DISCUSSION THREAD

**Discussion by CrystalWindSnake** [zauberzeug/nicegui#1894] [Ideas / Feature Requests]
Currently, nicegui requires `ui.tab_panels` to be used in conjunction with tabs. However, in quasar, tab panels can be used with other components. Here is an example of using radio buttons with tab panels.

```python

def on_radio_change(e):
    panels._props["model-value"] = e.value
    panels.update()


ui.radio(["mails", "alarms"], on_change=on_radio_change)

with ui.element("q-tab-panels").props("animated").classes(
    "shadow-2 rounded-borders w-full"
) as panels:
    with ui.element("q-tab-panel").props('name="mails"'):
        ui.label("mails page")

    with ui.element("q-tab-panel").props('name="alarms"'):
        ui.label("alarms page")

```
it is currently not possible to directly use ui.tab_panels because its first parameter must be an instance of ui.tabs.

**Comment by falkoschindler** [comment]
That's a great idea, @CrystalWindSnake!
I wasn't aware that tab panels can live without tabs.
But we probably only need to make `tabs` optional and make the `bind_value` call conditional.
Would you like to create a pull request? 🙂

**Comment by CrystalWindSnake** [comment]
I will gladly create a pull request

**Reply by falkoschindler** [reply]
PR #1897 has been released in version 1.4.2.
